### ISSUE TITLE
Implement permanent live video

### ISSUE THREAD

**Issue by Chocobozzz** [Chocobozzz/PeerTube#3389]
For now, when the user creates a live video, this video has its own stream key. When the live ends, the user cannot stream into this live anymore. It has some drawbacks:

 * The stream key changes every time the user creates a live, so it could be annoying to reconfigure the stream software
 * The user does not have a single URL that can be shared just one time and cannot say "I'll stream in this live every day at 8PM"
 * If there is a network issue, the user may not be able to restream in the same live video (because it ended)

A *permanent live video* is a live where the replay cannot be saved (for now, because it replaces the current live) but where the user can stream into it several times. The URL and the stream key do not change. It's like a twitch live, that the user can create according to his own desire.

**Comment by Poslovitch** [comment]
I haven't looked deeper into how Livestreams work on PeerTube, but I assume a video channel can only have a single live running at the same time. If that's the case, what about creating a dedicated "link" for this?

As an example, if I'd livestream on my "Poslovitch Live" channel, the link would be: `https://example.com/livestream/poslovitch_live`.

This, however, might not take content federation into account properly. Any thoughts?

**Comment by Chocobozzz** [comment]
Implemented in https://github.com/Chocobozzz/PeerTube/commit/bb4ba6d94c5051fdd665ebe63fffcc105778b8be

**Comment by Booteille** [comment]
Have you already ideas about how to implement replay for permanent lives?

I was thinking about generating a video with a generic name and informations, then notify the user it needs to be edited.

**Comment by Chocobozzz** [comment]
> I was thinking about generating a video with a generic name and informations, then notify the user it needs to be edited.

Yep something like that. I created an issue: https://github.com/Chocobozzz/PeerTube/issues/3396
